### ISSUE TITLE
Sources in 'dotnet list package' need to include package source mapping details

### ISSUE THREAD

**Issue by erdembayar** [NuGet/Home#11557]
Currently it only displays sources but if package source mapping is enabled then we need to inform about it for precision.
For example: if someone is checking for dependency confusion attack then it helps to narrow down there it came from.

https://github.com/NuGet/Home/pull/11446/files#r798101062
Given GH is not good at keep track of comment/discussion I'll include screenshot too.

<img width="535" alt="image" src="https://user-images.githubusercontent.com/8766776/152261460-0af412c9-276f-4ae7-9b7b-87879a2cc0ad.png">

**Comment by nkolev92** [comment]
The command in quesiton here is `list package`. 
We have an equivalent `list sources` command. 

We'll likely have an equivalent `list source-mappings`. 

Curious what @JonDouglas thinks, but my proposal would be that we don't add it right now and close this for nwo.

**Comment by erdembayar** [comment]
> The command in quesiton here is `list package`. We have an equivalent `list sources` command.
> 
> We'll likely have an equivalent `list source-mappings`.
> 
> Curious what @JonDouglas thinks, but my proposal would be that we don't add it right now and close this for nwo.

If we dig down there're so many edge cases.

**Comment by nkolev92** [comment]
Can you clarify what you mean by edge cases?

**Comment by erdembayar** [comment]
> Can you clarify what you mean by edge cases?

I just mean something like this. `Package source mapping  +  some scenario`.

**Comment by JonDouglas** [comment]
Yes I think it's a good issue to track. As Nikolche mentions, there will likely be other commands that give different outputs closer to what you might see in a nuget.config.

This is a good start to track the addition of seeing sources clearly defined in CLI. It may turn out similar to transitive dependencies and where they came from. i.e.

![image](https://user-images.githubusercontent.com/1288848/152343610-663e1a87-b319-46ec-83b9-169ea819a67f.png)

But with source used to resolve or in order if mapped multiple.

**Comment by nkolev92** [comment]
My suggestion actually was to close it, because this is the last thing we'd do, but don't feel too strongly :D
